lib/concurrent/atomic/semaphore.rb

```diff
@@ -141,12 +145,13 @@ def try_acquire_now(permits)
     end
 
     def try_acquire_timed(permits, timeout)
-      remaining = Condition::Result.new(timeout)
-      while !try_acquire_now(permits) && remaining.can_wait?
-        @condition.signal
-        remaining = @condition.wait(@mutex, remaining.remaining_time)
-      end
-      remaining.can_wait? ? true : false
+      ns_wait_until(timeout) { try_acquire_now(permits) }
```

Comment:
@pitr-ch Can you please verify that I'm using `ns_wait_until` correctly here? And if I am, I believe I can delete the tests [here](https://github.com/ruby-concurrency/concurrent-ruby/blob/more-synchronization/spec/concurrent/atomic/semaphore_spec.rb#L105) since spurious wakeups are now handled by the base class. Is that correct?